_posts/2015-07-09-upgrading-makara.md

```diff
@@ -0,0 +1,215 @@
+
+```javascript
+'use strict';
+
+var path = require('path');
+
+module.exports = function dustjs(grunt) {
+    // Load task
+    grunt.loadNpmTasks('grunt-dustjs');
+
+    // Options
+    return {
+        build: {
+            files: [
+                {
+                    expand: true,
+                    cwd: 'public/templates/',
+                    src: '**/*.dust',
+                    dest: '.build/templates',
+                    ext: '.js'
+                }
+            ],
+            options: {
+                fullname: function (filepath) {
+                    return path.relative('public/templates/', filepath).replace(/[.]dust$/, '');
+                }
+            }
+        }
+    };
+};
+```
+
+And the `tasks/localizr.js` isn't needed at all:
+
+```bash
+git rm tasks/localizr.js
+```
+
+Finally, make sure that each template has a corresponding `.properties` file with its localized strings, or, if that makes a lot of empty or duplicate content bundles, add `{@useContent}` helpers to your templates like so:
+
+```javascript
+    "dust": {
+        "helpers": [
+            {
+                "module": "dust-makara-helpers",
+                "arguments": [ { "autoloadTemplateContent": false } ]
+            }
+        ]
+    },
+```
+
+And add `{@useContent}` helpers to each template to select which bundles to load from. See [makara] for more information there.
+
+One last thing: The integration for where these components get locale information from has solidified. Instead of `res.locals.context.locality`, your app should set up `res.locals.locale` with a BCP47-compatible object or string. If you want a module that does this based on the browser's `Accept-Language` headers, check out [express-bcp47].
+
+
+## Browser application changes
+
+If you do all of the above and remove localizr, and your application requests pre-compiled templates from the browser, there are some important differences to be aware of and additional steps to take.
+
+kraken-devtools needs a copy of `dustjs-linkedin@^2.7.2` in your main project, to match the version that adaro requires:
+
+```bash
+npm install --save dustjs-linkedin@^2.7.2
+```
+
+Remove the `"i18n"` block from the `"template"` section of the devtools middleware configuration in development.json. The new 
+configuration will be:
+
+```javascript
+                        "template": {
+                            "module": "kraken-devtools/plugins/dustjs",
+                            "files": "/templates/**/*.js",
+                            "base": "templates"
+                        },
+```
+
+Since compiled templates are no longer placed into locale-specific directories, a request such as `/myapp/templates/FR/fr/index.js` will no longer resolve to a file. The new request will be `/myapp/templates/index.js`. Adjust your client accordingly, and use [dust-usecontent-helpers] and [dust-message-helper] or [dust-intl] to load the content for your templates.
```

Comment:
Not a requirement for publishing this. I will follow that use case down to the nuts and bolts though because I started down that road already. If I can get it together I'll submit a blog post for that next week.